moz_sql_parser/formatting.py

```diff
@@ -240,7 +247,8 @@ def orderby(self, json):
 
     def limit(self, json):
         if 'limit' in json:
-            return 'LIMIT {0}'.format(self.dispatch(json['limit']))
+            if json['limit']:
```

Comment:
All tests pass without this check.  What was your intent here?

Reply:
Good catch. It turns out that this might be a bug that needs to be fixed on the parser side instead of the formatter. 

The current code parses the following query
```
SELECT City FROM Customers UNION SELECT City FROM Suppliers ORDER BY City
```
into
```
{
    "from": {
        "union": [
            {
                "select": {
                    "value": "City"
                },
                "from": "Customers"
            },
            {
                "select": {
                    "value": "City"
                },
                "from": "Suppliers"
            }
        ]
    },
    "orderby": {
        "value": "City"
    },
    "limit": {}
}
```
The empty "limit" clause should not be present in the parse tree. And applying the formatting function on this parse tree returns the query
```
SELECT City FROM Customers UNION SELECT City FROM Suppliers ORDER BY City LIMIT
```
which is non-syntactical.

The solution is to either implement a fix in the formatter (which is what I did) or change the parser to not generating the empty "limit" clause.

Reply:
I pushed a fix for the `"limit":{}` problem.